src/main/php/xp/web/srv/Protocol.class.php

```diff
@@ -0,0 +1,107 @@
+<?php namespace xp\web\srv;
+
+use Generator;
+use peer\server\{AsyncServer, ServerProtocol};
+
+/** Multiplex protocol */
+class Protocol implements ServerProtocol {
+  private $protocols= [];
+
+  /** Creates a new instance of this multiplex protocol */
+  public static function multiplex(): self {
+
+    // Compatibility with older xp-framework/networking libraries, see issue #79
+    // Unwind generators returned from handleData() to guarantee their complete
+    // execution.
+    if (class_exists(AsyncServer::class, true)) {
```

Comment:
We already have a minimum requirement of `^10.1` for the networking library, which is exactly the version containing the async server, see https://github.com/xp-framework/networking/releases/tag/v10.1.0